File: cpp/program/perahuBermotor.cpp
#pragma once
#include "kendaraanBermotor.cpp"

class PerahuBermotor : public KendaraanBermotor {
private:
    // atribut
    string tipePerahu;
    int dayaApung;

    Mesin mesin;
    Transmisi transmisi;
    TangkiBahanBakar tangkiBahanBakar;

public:
    // constructor default
    PerahuBermotor() : KendaraanBermotor() {}

    // constructor parameter
    PerahuBermotor(string merk, string model, string tipePerahu, int dayaApung,
                   Mesin mesin, Transmisi transmisi, TangkiBahanBakar tangkiBahanBakar)
        : KendaraanBermotor(merk, model, mesin, transmisi, tangkiBahanBakar) {
        this->tipePerahu = tipePerahu;
        this->dayaApung = dayaApung;
        this->mesin = mesin;
        this->transmisi = transmisi;
        this->tangkiBahanBakar = tangkiBahanBakar;
    }

    // setter
    void setTipePerahu(string tipePerahu) {
        this->tipePerahu = tipePerahu;
    }

    void setDayaApung(int dayaApung) {
        this->dayaApung = dayaApung;
    }

    void setMesin(Mesin mesin) {
        this->mesin = mesin;
    }

    void setTransmisi(Transmisi transmisi) {
        this->transmisi = transmisi;
    }

    void setTangkiBahanBakar(TangkiBahanBakar tangkiBahanBakar) {
        this->tangkiBahanBakar = tangkiBahanBakar;
    }

    // getter
    string getTipePerahu() {
        return this->tipePerahu;
    }

    int getDayaApung() {
        return this->dayaApung;
    }

    Mesin getMesin() {
        return this->mesin;
    }

    Transmisi getTransmisi() {
        return this->transmisi;
    }

    TangkiBahanBakar getTangkiBahanBakar() {
        return this->tangkiBahanBakar;
    }

    // destructor
    ~PerahuBermotor() {}
};

File: java/program/PerahuBermotor.java
// PerahuBermotor.java
public class PerahuBermotor extends KendaraanBermotor {
    // atribut
    private String tipePerahu;
    private int dayaApung;

    // Constructor default
    public PerahuBermotor() {
        super();
    }

    // Constructor parameter
    public PerahuBermotor(String merk, String model, String tipePerahu, int dayaApung,
                          Mesin mesin, Transmisi transmisi, TangkiBahanBakar tangkiBahanBakar) {
        super(merk, model, mesin, transmisi, tangkiBahanBakar);
        this.tipePerahu = tipePerahu;
        this.dayaApung = dayaApung;
    }

    // Setter
    public void setTipePerahu(String tipePerahu) {
        this.tipePerahu = tipePerahu;
    }

    public void setDayaApung(int dayaApung) {
        this.dayaApung = dayaApung;
    }

    // Getter
    public String getTipePerahu() {
        return tipePerahu;
    }

    public int getDayaApung() {
        return dayaApung;
    }
}
